android/app/src/main/kotlin/coop/polypoly/polypod/polyNav/PolyNav.kt

```diff
@@ -65,15 +69,17 @@ open class PolyNav(
                     )
             }
         }
-        contentResolver?.openInputStream(importedUrl).use { inputStream ->
-            if (inputStream == null) {
-                throw Error("File copy error")
+        coroutineScope.launch {
+            contentResolver?.openInputStream(importedUrl).use { inputStream ->
+                if (inputStream == null) {
+                    throw Error("File copy error")
+                }
+                val newId = UUID.randomUUID().toString()
+                val fs = Preferences.getFileSystem(context).toMutableMap()
+                fs[fsPrefix + newId] = fileName
```

Comment:
This somehow changed the API of the imported files. I wonder how this could be made more visible, because it's going to have implications all across the feature. I know you've only moved around it here, but it's now even more indented and maybe more difficult to find. Maybe putting it into a function with a nice name like mapToUniqueName or something like that would make it clearer.

Reply:
I am not quite following here, sorry. The only thing that's changed for the API is that it is now returning before finishing the import. But the ID is the same as it used it. The proposed `mapToUniqueName` would take in a path and return a UUID? Isn't that essentially what `randomUUID()` is?

Reply:
No, a name and return an URI. If we decide to change the polypod://UUID prefix to poly://pod.UUID we should simply look at that function.

Reply:
Hm, I'm thinking this is a symptom of having `polypod://` hard coded. Logically, `polypod://foobar` means that we have a directory `foobar` at the root, so I think we may want to have some path building primitives. Pseudo code:

```
val path = polyPath.root.add(newId)
doSomethingWithFileSystemFunctions(path.localPath)
return path.featureFacingUrl
```